packages/form/addon/components/cf-field/input/files.js

```diff
@@ -41,48 +41,47 @@ export default class CfFieldInputFilesComponent extends Component {
     const fileList = [...(this.files || [])];
 
     // unwrap files from FileList construct
-    const newFiles = [];
-    for (let i = 0; i < target.files.length; i++) {
-      newFiles.push({ name: target.files[i].name });
-      if (
-        !fileList.find((existing) => existing.name === target.files[i].name)
-      ) {
-        fileList.push({ name: target.files[i].name });
-      }
-    }
+    const newFiles = Array.from(target.files).map((file) => ({
+      name: file.name,
+      value: file,
+    }));
+
     if (!newFiles) {
       return;
     }
 
-    // trigger save action for deduped file list of old and new files reduces properties
+    newFiles
+      .filter(
+        (file) => !fileList.find((existing) => existing.name === file.name)
```

Comment:
Consider using presence of `id` attribute to figure out if a file is new or not

Reply:
As there is no need to restrict the behavior in terms of duplicated file upload, I will leave out the filtering per se. As this can lead to duplicated names in the file list I have to refactor the component a little bit for deleting and downloading (`file.name` was used as identifier).